relay-general/src/protocol/stacktrace.rs

```diff
@@ -350,6 +362,123 @@ pub struct RawStacktrace {
     pub other: Object<Value>,
 }
 
+/// Controls the mechanism by which the `instruction_addr` of a [`Stacktrace`] [`Frame`] is adjusted.
+///
+/// The adjustment tries to transform *return addresses* to *call addresses* for symbolication.
+/// Typically, this adjustment needs to be done for all frames but the first, as the first frame is
+/// usually taken directly from the cpu context of a hardware exception or a suspended thread and
+/// the stack trace is created from that.
+///
+/// When the stack walking implementation truncates frames from the top, `"all"` frames should be
+/// adjusted. In case the stack walking implementation already does the adjustment when producing
+/// stack frames, `"none"` should be used here.
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+#[cfg_attr(feature = "jsonschema", derive(schemars::JsonSchema))]
+#[cfg_attr(feature = "jsonschema", schemars(rename_all = "snake_case"))]
+pub enum InstructionAddrAdjustment {
+    /// The default. Applies a heuristic based on other event / exception attributes.
+    Auto,
+
+    /// All but the first frame needs to be adjusted. The first frame's address is not a *return address*,
+    /// but points directly to the faulty instruction.
+    AllButFirst,
+
+    /// All frames should be adjusted, for example because the stack walking implementation truncated
+    /// frames from the top of the stack, and all remaining frames' addresses are *return addresses*.
+    All,
+
+    /// The stack walking implementation already provides correct addresses and no adjustment should
+    /// be performed when symbolicating.
+    None,
+}
+
+/// An error used when parsing `Level`.
+#[derive(Debug)]
+pub struct ParseAdjustmentError;
+
+impl fmt::Display for ParseAdjustmentError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "invalid instruction_addr_adjustment")
+    }
+}
+
+impl std::error::Error for ParseAdjustmentError {}
+
+impl FromStr for InstructionAddrAdjustment {
+    type Err = ParseAdjustmentError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "auto" => Ok(Self::Auto),
+            "all_but_first" => Ok(Self::AllButFirst),
+            "all" => Ok(Self::All),
+            "none" => Ok(Self::None),
+            _ => Err(ParseAdjustmentError),
+        }
+    }
+}
+
+impl fmt::Display for InstructionAddrAdjustment {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let s = match self {
+            InstructionAddrAdjustment::Auto => "auto",
+            InstructionAddrAdjustment::AllButFirst => "all_but_first",
+            InstructionAddrAdjustment::All => "all",
+            InstructionAddrAdjustment::None => "none",
+        };
+        f.write_str(s)
+    }
+}
```

Comment:
`derive_fromstr_and_display` might be useful here, see for example https://github.com/getsentry/relay/blob/0df83a57639e00fa89a51c8e05123d7bba7c497d/relay-general/src/protocol/security_report.rs#L522